kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/OrderPreservingBase64Encoder.java

```diff
@@ -0,0 +1,44 @@
+package dev.responsive.kafka.internal.db.mongo;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.codec.binary.Base64;
+
+public class OrderPreservingBase64Encoder {
+  private static final Base64 encoder = Base64.builder()
+      .setEncodeTable(
+          encoderCharset("+/0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"))
+      .get();
```

Comment:
since this is a custom b64 encoding anyway (meaning we can't use online tools to decode it), I'm wondering why we're limiting ourselves to just b64? couldn't we do the same thing with a wider charset?

Reply:
Mongo strings are UTF-8 encoded. So the only charset size that would actually reduce overhead is 7 bits (basically full ascii). Larger bit-sizes would use enough additional bits to do the utf-8 encoding that the overhead is actually worse (for example, to encode 8-11 bit characters in utf-8 you have to use 16 bits). I'm hesitant to use the full ascii charset because
1. I'm not sure how mongo's find will or wont interpret the special characters.
2. We'd have to write the encoding ourself, which isn't that hard, but its nice to just have a library to use instead.